block/retriever.go

```diff
@@ -99,19 +99,54 @@ func (m *Manager) processNextDABatch(ctx context.Context, daHeight uint64) error
 	return nil
 }
 
-func (m *Manager) fetchBatch(daHeight uint64) (da.ResultRetrieveBatch, error) {
+func (m *Manager) fetchBatch(daMetaData *da.DAMetaData) (da.ResultRetrieveBatch, error) {
 	var err error
-	batchRes := m.retriever.RetrieveBatches(daHeight)
+	availRes := m.retriever.CheckBatchAvailability(daMetaData)
+
+	switch availRes.Code {
```

Comment:
imo leaving the code each status should indiciate a different handling. i.e `success` means on thing, `errorGettingProof` can be another and `ProofInvalid` can be another one but if they have the same handling route than the status code a bit looses the point IMO. If you want to add extra info to each status you can add it in the `message` field.